p2p/src/net/mock/transport/tcp/mod.rs

```diff
@@ -40,58 +43,75 @@ use crate::{
     P2pError, Result,
 };
 
+use self::adapter::StreamAdapter;
+
 #[derive(Debug)]
-pub struct TcpMockTransport {}
+pub struct TcpMockTransport<E>(std::marker::PhantomData<E>);
```

Comment:
Why are we owning `PhantomData<E>` instead of E?

Reply:
Not sure why we would want to own `StreamAdapter`, `StreamAdapter` trait has only static methods right now.